src/Vision/V1/ImageAnnotatorClient.php

```diff
@@ -30,13 +30,201 @@
+     *     $image = $client->createImageObject($imageResource);
+     *     $response = $client->faceDetection($request);
+     * } finally {
+     *     $client->close();
+     * }
+     * ```
+     *
+     * ```
+     * // Create an Image object from raw image data.
+     * $client = new ImageAnnotatorClient();
+     * try {
+     *     $imageData = file_get_contents('path/to/image.jpg');
+     *     $image = $client->createImageObject($imageData);
+     *     $response = $client->faceDetection($request);
+     * } finally {
+     *     $client->close();
+     * }
+     * ```
+     *
+     * ```
+     * // Create an Image object from a Google Cloud Storage Uri.
+     * $client = new ImageAnnotatorClient();
+     * try {
+     *     $imageUri = "gs://my-bucket/image.jpg";
+     *     $image = $client->createImageObject($imageUri);
+     *     $response = $client->faceDetection($request);
+     * } finally {
+     *     $client->close();
+     * }
+     * ```
+     *
+     * @param  resource|string $imageInput An image to configure with
+     *         the given settings. This parameter will accept a resource, a
+     *         string of bytes, or the URI of an image in a publicly-accessible
+     *         web location.
+     * @return Image
+     * @throws InvalidArgumentException
+     */
+    public function createImageObject($imageInput)
+    {
+        return $this->createImageHelper(Image::class, ImageSource::class, $imageInput);
+    }
+
+    /**
+     * Run image detection and annotation for an image.
+     *
+     * Sample code:
+     * ```
+     * $imageAnnotatorClient = new ImageAnnotatorClient();
+     * try {
+     *     $imageResource = fopen('path/to/image.jpg', 'r');
+     *     $image = $this->createImageObject($imageResource);
+     *     $feature = new Feature();
+     *     $feature->setType(Feature_Type::FACE_DETECTION);
+     *     $features = [$feature];
+     *     $request = new AnnotateImageRequest();
+     *     $request->setImage($image);
+     *     $request->setFeatures($features);
+     *     $response = $imageAnnotatorClient->annotateImage($request);
+     * } finally {
+     *     $imageAnnotatorClient->close();
+     * }
+     * ```
+     *
+     * @param AnnotateImageRequest $request      An image annotation request.
+     * @param array                $optionalArgs {
+     *                                           Optional.
+     *
+     *     @type RetrySettings|array $retrySettings
+     *          Retry settings to use for this call. Can be a
+     *          {@see Google\ApiCore\RetrySettings} object, or an associative array
+     *          of retry settings parameters. See the documentation on
+     *          {@see Google\ApiCore\RetrySettings} for example usage.
+     * }
+     *
+     * @return AnnotateImageResponse
+     *
+     * @throws ApiException if the remote call fails
+     * @experimental
+     */
+    public function annotateImage($request, $optionalArgs = [])
```

Comment:
I would definitely appreciate some feedback on this method. I have copied python's approach of accepting a `request` object. However, my initial thought was to flatten this method, so as to accept an `Image`, a `Feature` list, and an optional `ImageContext`. WDYT?

Reply:
I agree that it would be good to flatten it. One less thing for users to worry about.